src/Chainweb/Pact/Backend/Compaction.hs

```diff
@@ -626,3 +635,56 @@ compactMain = do
     fromTextSilly t = case fromText t of
       Just a -> a
       Nothing -> error "fromText failed"
+
+getLatestPactState :: Database -> IO (Map Text [PactRow])
+getLatestPactState db = do
+  let checkpointerTables = ["BlockHistory", "VersionedTableCreation", "VersionedTableMutation", "TransactionIndex"]
+  let compactionTables = ["CompactGrandHash", "CompactActiveRow"]
+  let excludeThese = checkpointerTables ++ compactionTables
+  let fmtTable x = "\"" <> x <> "\""
+
+  tables <- fmap sortedTableNames $ do
+    let qry =
+          "SELECT name FROM sqlite_schema \
+          \WHERE \
+          \  type = 'table' \
+          \AND \
+          \  name NOT LIKE 'sqlite_%'"
```

Comment:
I like that we have a blacklist for non-compacted tables rather than the other way around.

Reply:
So, you are saying, you prefer the code as written? Or are you suggesting a change?